kotest-extensions/src/jvmMain/kotlin/io/kotest/extensions/clock/TestClock.kt

```diff
@@ -0,0 +1,49 @@
+package io.kotest.extensions.clock
+
+import java.time.Clock
+import java.time.Instant
+import java.time.ZoneId
+import java.time.ZoneOffset
+import kotlin.time.Duration
+
+/**
+ * A mutable [Clock] that supports millisecond precision.
+ */
+class TestClock(
+   private var instant: Instant,
+   private val zone: ZoneId,
+) : Clock() {
+
+   companion object {
+      fun utc(instant: Instant) = TestClock(instant, ZoneOffset.UTC)
+   }
+
+   override fun instant(): Instant = instant
+
+   override fun withZone(zone: ZoneId): Clock {
+      return TestClock(instant, zone)
+   }
+
+   override fun getZone(): ZoneId = zone
+
+   /**
+    * Sets the instant in this test clock to the given value.
+    */
+   fun setInstant(instant: Instant) {
+      this.instant = instant
```

Comment:
do we need this thing to be thread safe, and use `AtomicReference`?

Reply:
That's up to the caller. I think its best to keep this simple.